server/remote/user.go

```diff
@@ -16,3 +20,9 @@ type MailboxSettings struct {
 	TimeZone     string       `json:"timeZone"`
 	WorkingHours WorkingHours `json:"workingHours"`
 }
+
+type UserTokenHelpers struct {
+	CheckUserStatus      func(mattermostUserID string) bool
+	ChangeUserStatus     func(err error, mattermostUserID string)
+	RefreshAndStoreToken func(token *oauth2.Token, oconf *oauth2.Config, mattermostUserID string) (*oauth2.Token, error)
+}
```

Comment:
Just a thought, we could make this an interface and pass the store itself as an arg instead of creating this struct. The remote package would unknowingly be using the store in that case. The previous coupling would still be removed since there is no direct reference to the store package.

Reply:
If you are able to, I would prefer to use an interface and pass the store as mentioned above. This is the same pattern as passing the `bot` as both `Logger` and `Poster`

Reply:
Using structs for this sort of thing is an anti-pattern in Go, especially when something (in this case the store) implements all of the methods, and can be treated as an interface. This allows for seamless mocking etc.